src/components/time-entries/TimeEntries.tsx

```diff
@@ -1,109 +1,100 @@
-
-        const dateStringToday = today.toLocaleDateString("nl-NL");
-        const dateStringYesterday = yesterday.toLocaleDateString("nl-NL");
-
-        const dateObject = new Date(timeEntry.startTime);
-        const currentDate = dateObject.toLocaleDateString("nl-NL");
-
-        const nextTimeEntry = timeEntries?.[index + 1];
-        const nextDateObject = new Date(nextTimeEntry?.startTime);
-        const nextDate = nextDateObject?.toLocaleDateString("nl-NL");
-
-        const previousTimeEntry = timeEntries?.[index - 1];
-        const previousDateObject = new Date(previousTimeEntry?.startTime);
-        const previousDate = previousDateObject?.toLocaleDateString("nl-NL");
-
-        function isSameDay() {
-          if (currentDate === dateStringToday) {
-            return "(Today)";
+      {timeEntries
+        .filter((timeEntry) => timeEntry.client.toLowerCase().includes(searchInput.toLowerCase()))
+        .map((timeEntry, index) => {
+          const startTimeObject = new Date(timeEntry.startTime);
+          const startTime = startTimeObject.toLocaleTimeString("nl-NL", {
+            hour: "2-digit",
+            minute: "2-digit",
+          });
+
+          const endTimeObject = new Date(timeEntry.endTime);
+          const endTime = endTimeObject.toLocaleTimeString("nl-NL", {
+            hour: "2-digit",
+            minute: "2-digit",
+          });
+
+          const today = new Date();
+          const yesterday = new Date(today);
+
+          yesterday.setDate(yesterday.getDate() - 1);
+
+          const dateStringToday = today.toLocaleDateString("nl-NL");
+          const dateStringYesterday = yesterday.toLocaleDateString("nl-NL");
+
+          const dateObject = new Date(timeEntry.startTime);
+          const currentDate = dateObject.toLocaleDateString("nl-NL");
+
+          const nextTimeEntry = timeEntries?.[index + 1];
+          const nextDateObject = new Date(nextTimeEntry?.startTime);
+          const nextDate = nextDateObject?.toLocaleDateString("nl-NL");
+
+          const previousTimeEntry = timeEntries?.[index - 1];
+          const previousDateObject = new Date(previousTimeEntry?.startTime);
+          const previousDate = previousDateObject?.toLocaleDateString("nl-NL");
+
+          function isSameDay() {
+            if (currentDate === dateStringToday) {
+              return "(Today)";
+            }
+
+            if (currentDate === dateStringYesterday) {
+              return "(Yesterday)";
+            }
+
+            return "";
           }
 
-          if (currentDate === dateStringYesterday) {
-            return "(Yesterday)";
+          if (index >= 1 && currentDate === previousDate) {
+            return (
+              <TimeEntry
+                client={timeEntry.client}
+                deleteTimeEntry={deleteEntry}
+                endTime={endTime}
+                id={timeEntry.id}
+                key={timeEntry.id}
+                lastEntry={currentDate !== nextDate}
```

Comment:
The component itself doesn't need to know which entry it is. It should only know what styling it needs to have 🧑‍🎨